tests/e2e/external/external_test.go

```diff
@@ -0,0 +1,195 @@
+// This test verifies that two nodes, which can't connect using the local network, are
+// able to still connect using the node-external-ip. In real life, node-external-ip
+// would be a public IP. In the test, we create two networks, one sets the node
+// internal-ip and the other sets the node-external-ip. Traffic is blocked on the former
+
+package validatenodeexternalip
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/k3s-io/k3s/tests/e2e"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+// Valid nodeOS: generic/ubuntu2004, opensuse/Leap-15.3.x86_64
+var nodeOS = flag.String("nodeOS", "generic/ubuntu2004", "VM operating system")
+var serverCount = flag.Int("serverCount", 1, "number of server nodes")
+var agentCount = flag.Int("agentCount", 1, "number of agent nodes")
+var hardened = flag.Bool("hardened", false, "true or false")
+
+// Environment Variables Info:
+// E2E_RELEASE_VERSION=v1.23.1+k3s1 or nil for latest commit from master
+
+type objIP struct {
+	name string
+	ipv4 string
+	ipv6 string
+}
```

Comment:
We should consolidate this under `testutils.go` and resue it across the `dualstack` and `external` test.

Reply:
good idea